### ISSUE TITLE
Strange 'yaml.reader.ReaderError: unacceptable character #x0082: special caracters are not allowed  in "<unicode string>", position 446' in modular openapi definitionh

### ISSUE THREAD

**Issue by rganowski** [jfinkhaeuser/prance#51]
A have a simple openapi definition prepared to present the problem, with a sinlge Polish letter: **ł**.

**openapi.yaml**:
```yaml
openapi: 3.0.2
info:
  title: title
  version: '1.0'
paths:
  /something/:
    post:
      summary: summary
      requestBody:
        content:
          application/json:
            schema:
              type: string
      responses:
        '400':
          description: description
          content:
            application/json:
              schema:
                type: object
                properties:
                  info:
                    type: array
                    items:
                      type: string
                      enum:
                        - ł
```
That snippet compiles as expected:
```sh
% prance compile --strict openapi.yaml all-in-one-openapi.yaml
Processing "openapi.yaml"...
 -> Resolving external references.
Validates OK as OpenAPI 3.0.2!
Output written to "all-in-one-openapi.yaml".
```

But I wanted to spit this definition into parts. So I prepared two another files: 

**openapi-main.yaml**:
```yaml
openapi: 3.0.2
info:
  title: title
  version: '1.0'
paths:
  /something/:
    $ref: 'openapi-part.yaml'
```

**openapi-part.yaml**:
```yaml
---
post:
  summary: summary
  requestBody:
    content:
      application/json:
        schema:
          type: string
  responses:
    '400':
      description: description
      content:
        application/json:
          schema:
            type: object
            properties:
              info:
                type: array
                items:
                  type: string
                  enum:
                    - ł
```
And now my problem arises:
```sh
% prance compile --no-strict openapi-main.yaml all-in-one-openapi.yaml

Processing "openapi-main.yaml"...
 -> Resolving external references.
Traceback (most recent call last):
  File "/home/xxx/.virtualenvs/prance/bin/prance", line 8, in <module>
    sys.exit(cli())
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/cli.py", line 103, in command_invoke
    original_invoke(ctx)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/cli.py", line 220, in compile
    __validate(parser, name)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/cli.py", line 54, in __validate
    parser.parse()
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/__init__.py", line 134, in parse
    self._validate()
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/__init__.py", line 271, in _validate
    resolver.resolve_references()
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/resolver.py", line 79, in resolve_references
    self.specs = self._resolve_partial(self.parsed_url, self.specs, ())
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/resolver.py", line 175, in _resolve_partial
    recursions)))
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/resolver.py", line 117, in _dereferencing_iterator
    ref_value = self._dereference(ref_url, obj_path, next_recursions)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/resolver.py", line 139, in _dereference
    contents = _url.fetch_url(ref_url, self.__reference_cache)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/url.py", line 215, in fetch_url
    result = parse_spec(content, url.path, content_type = content_type)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/formats.py", line 205, in parse_spec
    result, ctype, ext = parse_spec_details(spec_str, filename, **kwargs)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/formats.py", line 180, in parse_spec_details
    result = parser(spec_str)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/prance/util/formats.py", line 67, in __parse_yaml
    return yaml.safe_load(six.text_type(spec_str))
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/yaml/__init__.py", line 112, in load
    loader = Loader(stream)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/yaml/loader.py", line 34, in __init__
    Reader.__init__(self, stream)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/yaml/reader.py", line 74, in __init__
    self.check_printable(stream)
  File "/home/xxx/.virtualenvs/prance/lib/python3.7/site-packages/yaml/reader.py", line 144, in check_printable
    'unicode', "special characters are not allowed")
yaml.reader.ReaderError: unacceptable character #x0082: special characters are not allowed
  in "<unicode string>", position 432
```

In this case I can do some workarounds. Eg. remove three dashes (---) from the beginning of **openapi-part.yaml** file, which are not obligatory in such a situation, but it desn’t work in all my real cases. It’s also really amazing that I can leave those dashes in the file and remove its third line (summary:) and it works fine, as well. 

I can't discover any rules in this behavior.

I use prance==0.16.2 with python 3.7.5.

**Comment by jfinkhaeuser** [comment]
Thanks! I'll take a look.

**Comment by jfinkhaeuser** [comment]
Well the good news first: I can confirm this is happening.

The reason seems to be that the file encoding for the parts file is incorrectly detected as `iso-8859-2` when it is valid `utf-8`.

There is a heuristic in prance for detecting file encoding that's based on a bunch of backends, each of which should do better than prance itself, since they're specialized character encoding libraries. But still this fails.

As a workaround, I'm adding a CLI option for setting the file encoding, overriding detection. I'll also see if I can make the heuristic a little bit smarter.

**Comment by rganowski** [comment]
So excelent and so quick answer. Thank you very much.

**Comment by jfinkhaeuser** [comment]
Unfortunately, I'm not always that quick :)

**Comment by jfinkhaeuser** [comment]
Quick question, are you on Python3?

Python2 will no longer be maintained after the end of the year, so I've recently made changes to drop testing for it (it's still supported, I suppose). Would that impact you?

**Comment by rganowski** [comment]
Yes, yes, of course, as I pointed out at the end of my description:
> prance==0.16.2 python 3.7.5

Yet waiting with hope for walrus op. in 3.8 :)

Thx

**Comment by jfinkhaeuser** [comment]
Ah - I missed that, sorry. Ok, then I'll make this part of 0.17 rather than a patch release. Should be done soon either way.

**Comment by rganowski** [comment]
Once again thank you very much

**Comment by jfinkhaeuser** [comment]
Will be released with 0.17.0
